x-pack/platform/test/README.md

```diff
@@ -0,0 +1,204 @@
+
+- Must not rely on stateful-specific features (like Watcher, ML jobs with persistence)
+- Should use deployment-agnostic services and utilities
+- Must use SAML authentication (cookie header for internal APIs or API key for public APIs)
+
+### 4. Functional UI Tests - `/functional` and `/functional_*`
+
+**Purpose**: End-to-end UI tests using WebDriver for browser automation.
+
+**Locations**:
+
+- `x-pack/platform/test/functional/` - Main functional tests
+- `x-pack/platform/test/functional_*` - Specialized functional test suites
+
+**Types of functional test directories**:
+
+#### Core Functional Tests
+
+- **`/functional`** - Primary UI test suite
+- **`/functional_basic`** - Basic auth UI tests
+- **`/functional_embedded`** - Embedded Kibana scenarios
+- **`/functional_cors`** - Cross-origin resource sharing tests
+
+#### Specialized UI Test Suites
+
+- **`/functional_with_es_ssl`** - SSL-enabled Elasticsearch tests
+
+**Key functional testing concepts**:
+
+- **Page Objects**: Encapsulate UI interactions and selectors
+- **Services**: Provide reusable utilities (browser, retry, find)
+
+### 5. Serverless-Only Tests - `/serverless`
+
+> For detailed serverless testing documentation, see [x-pack/platform/test/serverless/README.md](./serverless/README.md)
+
+**Purpose**: Tests specifically designed for serverless Kibana deployments.
+
+**Location**: `x-pack/platform/test/serverless/`
+
+**When to use**:
+
+- Serverless-specific features and limitations
+- Platform functionality shared between different project types (Security, Observability, Search)
+- Serverless authentication and authorization patterns
+
+**Note**: Project-type specific scenarios should be located in respective solution test directories, not here.
+
+## Test Configuration and Services
+
+### Configuration Files
+
+Each test suite has its own `config.ts` file that defines:
+
+- Test server settings (Elasticsearch, Kibana)
+- Service providers and page objects
+- Test file inclusion patterns
+- Browser settings and capabilities
+- Authentication configuration
+
+### Common Services
+
+- **`supertest`** - HTTP API testing
+- **`es`** - Elasticsearch client
+- **`esArchiver`** - Test data management
+- **`kibanaServer`** - Kibana instance management
+- **`browser`** - WebDriver browser control
+- **`find`** - Element location utilities
+- **`testSubjects`** - Test-specific element selectors
+
+### Page Objects
+
+Reusable UI interaction patterns organized by application:
+
+- `PageObjects.common` - Shared navigation and utilities
+- `PageObjects.dashboard` - Dashboard-specific interactions
+- `PageObjects.discover` - Discover app interactions
+- `PageObjects.visualize` - Visualization creation and editing
+
+## Running Tests
+
+### Local Development
+
+```bash
+# Start test server
+node scripts/functional_tests_server.js --config x-pack/platform/test/functional/config.ts
+
+# Run tests
+node scripts/functional_test_runner.js --config x-pack/platform/test/functional/config.ts
```

Comment:
Did you intentionally pick a non-existing config file for this example code?

Reply:
regexp replacement 🙈 I will update all to existing ones